### DISCUSSION TITLE
Grafana dashboard shows cluster degraded but it doesn't seem so.

### DISCUSSION THREAD

**Discussion by alessandroros** [cloudnative-pg/cloudnative-pg#4487] [Q&A] [answered]
Hello,
I have a question.
Everything was working properly, so I tried to do a "failover test" :). After testing the failover I also "downsized" the cluster from 2 instance to 1 and viceversa for some times.
Right now I have 2 isntances 1 called prod-db-1 and the other called prod-db-6.

Here the status of my cluster
![CNPG status](https://github.com/cloudnative-pg/cloudnative-pg/assets/3214597/ea26064d-e2a7-4999-8865-f508c3373aff)

Here my grafana dashboard
![Grafana](https://github.com/cloudnative-pg/cloudnative-pg/assets/3214597/99ce2249-6c6c-48f9-94f1-e06ef97ba6cb)

If I look for the variables used by those 2 tiles I see:
![replication in recovery](https://github.com/cloudnative-pg/cloudnative-pg/assets/3214597/77bf8d06-d2ee-48b0-a87e-2d55cb6d6277)
Here it seems like db-6 is in recovery.

![replication_streaming_replica](https://github.com/cloudnative-pg/cloudnative-pg/assets/3214597/d5569022-bf82-4fc7-8748-5a40c3a6fa46)

![stat_archivier_seconds_since_last_archival](https://github.com/cloudnative-pg/cloudnative-pg/assets/3214597/90eb693b-ecf9-4276-9115-4b4f5cd1d5d4)

![replication_is_wal_receiver_up](https://github.com/cloudnative-pg/cloudnative-pg/assets/3214597/b0d2000e-f9ea-44f9-bdc3-4277ce7bf75c)
Here I see that the wal reciever is not up on the db-6, but I'm not sure if it's ok or not.

Do you see any issues? 
Thanks for anyone which could clarify something on these topics where I'm not totally familiar.

If I try to connect to DB6 (read only service) everything is fine, even if I retrigger a failover everything is ok (indipendently from the instance I'm killing). Any thoughts?

Alessandro

**Comment by fmiqbal** [comment, marked as answer]
I've also stumbled upon this, for the Replication status, its consist of this query:
```
max(cnpg_pg_replication_streaming_replicas{namespace=~"$namespace", pod=~"$instances"} - cnpg_pg_replication_is_wal_receiver_up{namespace=~"$namespace", pod=~"$instances"})
```
which according to the docs
> **cnpg_pg_replication_streaming_replicas** Number of streaming replicas connected to the instance
> **cnpg_pg_replication_is_wal_receiver_up** Whether the instance wal_receiver is up

and then the threshold

![image](https://github.com/cloudnative-pg/cloudnative-pg/assets/8409391/d634bce7-6a84-4310-a25e-fe59ba16d6b9)

I suspect the team consider the cluster healthy if it has 2 replica or more (1 primary, 2 standby minimum)

**Comment by wokalski** [comment]
in case you actually want 1 replica, here's a quick jq script to edit the dashboard:
```jq
            def transform_mappings:
              map(
                if .type == "value" then
                  .options |= with_entries(
                    select(.key != "1")
                  )
                elif .type == "range" then
                  .options.from = 1
                else
                  .
                end
              );

            (.panels | index(map(select(
              .type == "stat" 
              and ((.description // "") | contains("Cluster Replication Health"))
              and (.targets // [] | map(.expr) | any(contains("cnpg_pg_replication_streaming_replicas")))
            )) | .[0])) as $idx
            | if $idx == null then
                error("Could not find the target panel in the dashboard")
              else
                .panels[$idx].fieldConfig.defaults.mappings |= transform_mappings
              end
```
